### PR TITLE
 mkchromecast: add qt5.qtwayland 

### PR DESCRIPTION
**Pull request by SebTM**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/mkchromecast/default.nix [hunk 1]
```diff
@@ -46,7 +47,8 @@ python3Packages.buildPythonApplication rec {
     netifaces
     requests
     pyqt5
-  ] ++ lib.optionals enableSonos [ soco ]);
+  ] ++ lib.optionals enableSonos [ soco ])
+  ++ [ qt5.qtwayland ];
```
Comment (natsukium) [inline pkgs/applications/networking/mkchromecast/default.nix:51]: I think `qtwayland` should go to `buildInputs` and is only needed by Linux.

File: pkgs/applications/networking/mkchromecast/default.nix [hunk 2]
```diff
@@ -27,17 +28,18 @@ let packages = [
 ] ++ lib.optionals stdenv.isLinux [ pulseaudio ];
 
 in
-python3Packages.buildPythonApplication rec {
+python3Packages.buildPythonApplication {
```
Comment (SebTM) [inline pkgs/applications/networking/mkchromecast/default.nix:31]: Seems unused from lsp

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 238002` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mkchromecast</li>
    <li>mkchromecast.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM

File: pkgs/applications/networking/mkchromecast/default.nix [hunk 3]
```diff
@@ -13,6 +13,7 @@
 , opusTools
 , gst_all_1
 , enableSonos ? true
+, qt5
```
Comment (wineee) [inline pkgs/applications/networking/mkchromecast/default.nix:16]:
```suggestion
, qtwayland
```

use `libsForQt5.callPackage` in `pkgs/top-level/all-packages.nix`

File: pkgs/applications/networking/mkchromecast/default.nix [hunk 4]
```diff
@@ -27,17 +28,18 @@ let packages = [
...
 
-  src = fetchFromGitHub rec {
+  src = fetchFromGitHub {
     owner = "muammar";
     repo = "mkchromecast";
     rev = "0de9fd78c4122dec4f184aeae2564790b45fe6dc";
     sha256 = "sha256-dxsIcBPrZaXlsfzOEXhYj2qoK5LRducJG2ggMrMMl9Y=";
   };
 
+  buildInputs = lib.optional stdenv.isLinux qt5.qtwayland;
```
Comment (wineee) [inline pkgs/applications/networking/mkchromecast/default.nix:42]:
```suggestion
  buildInputs = lib.optional stdenv.isLinux qtwayland;
```

**Review by wineee — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/mkchromecast/default.nix (hunk 3 above)
Comment (SebTM) [reply pkgs/applications/networking/mkchromecast/default.nix]: `libsForQt5.callPackage` is already used just didn't know ✌🏻

**Comment by Janik-Haag** [comment]
please squash the commits so you only have the add qtwayland one.

**Comment by SebTM** [comment]
Thanks for the feedback, reflected it in the code-changes.

**Review by wineee — APPROVED** [review]

**Review by Janik-Haag — APPROVED** [review]
LGTM; will merge once ci passes

**Comment by Janik-Haag** [comment]
@ofborg evaluate

**Comment by Janik-Haag** [comment]
@ofborg eval

**Comment by Janik-Haag** [comment]
@ofborg build mkchromecast

**Comment by Janik-Haag** [comment]
Seems like the ci is borked :(

**Pull request merged**